engine/src/main/java/org/terasology/entitySystem/entity/internal/PojoEntityManager.java

```diff
@@ -606,29 +530,20 @@ public void saveComponent(long entityId, Component component) {
      * Implementation
      */
 
-    protected void assignToPool(EntityRef ref, EngineEntityPool pool) {
-        //Todo: job for the sector manager?
-        poolMap.put(ref.getId(), pool);
+    private EngineEntityPool getEntityPool(long id) {
+        EngineEntityPool pool = poolMap.get(id);
+        if (pool == null) {
+                logger.error("Entity {} doesn't have an assigned pool", id);
+                return globalPool;
+        } else {
+            return pool;
+        }
     }
 
     protected void assignToPool(long entityId, EngineEntityPool pool) {
```

Comment:
Yep, I think that will be needed to move entities between scopes, and to allow the sector manager to split up/join the sectors.

Should it send an event when entities are moved between caches?